tensorflow_addons/metrics/tests/cohens_kappa_test.py

```diff
@@ -18,180 +18,175 @@
-            [kp_obj1, kp_obj2, kp_obj3], [-0.3333333, -0.52380952, -0.72727272]
-        )
-
-    def test_kappa_with_sample_weights(self):
-        actuals = [4, 4, 3, 3, 2, 2, 1, 1]
-        preds = [1, 2, 4, 1, 3, 3, 4, 4]
-        weights = [1, 1, 2, 5, 10, 2, 3, 3]
-        actuals = tf.constant(actuals, dtype=tf.int32)
-        preds = tf.constant(preds, dtype=tf.int32)
-        weights = tf.constant(weights, dtype=tf.int32)
-
-        # Initialize
-        kp_obj1, kp_obj2, kp_obj3 = self.initialize_vars()
-
-        # Update
-        self.update_obj_states(kp_obj1, kp_obj2, kp_obj3, actuals, preds, weights)
-
-        # check results
-        self.check_results(
-            [kp_obj1, kp_obj2, kp_obj3], [-0.25473321, -0.38992332, -0.60695344]
-        )
+    # Check save and restore config
+    kb_obj2 = CohenKappa.from_config(kp_obj.get_config())
+    assert kb_obj2.name == "cohen_kappa"
+    assert kb_obj2.dtype == tf.float32
+    assert kp_obj.num_classes == 5
 
-    def test_kappa_reset_states(self):
-        # Initialize
-        kp_obj1, kp_obj2, kp_obj3 = self.initialize_vars()
 
-        # reset states
-        self.reset_obj_states(kp_obj1, kp_obj2, kp_obj3)
+def initialize_vars():
+    kp_obj1 = CohenKappa(num_classes=5, sparse_labels=True)
+    kp_obj2 = CohenKappa(num_classes=5, sparse_labels=True, weightage="linear")
+    kp_obj3 = CohenKappa(num_classes=5, sparse_labels=True, weightage="quadratic")
 
-        # check results
-        self.check_results([kp_obj1, kp_obj2, kp_obj3], [0.0, 0.0, 0.0])
+    return kp_obj1, kp_obj2, kp_obj3
 
-    def test_large_values(self):
-        y_true = [1] * 10000 + [0] * 20000 + [1] * 20000
-        y_pred = [0] * 20000 + [1] * 30000
 
-        y_true = tf.convert_to_tensor(y_true)
-        y_pred = tf.convert_to_tensor(y_pred)
+def update_obj_states(obj1, obj2, obj3, actuals, preds, weights):
+    obj1.update_state(actuals, preds, sample_weight=weights)
+    obj2.update_state(actuals, preds, sample_weight=weights)
+    obj3.update_state(actuals, preds, sample_weight=weights)
 
-        obj = CohenKappa(num_classes=2)
-        self.evaluate(tf.compat.v1.variables_initializer(obj.variables))
 
-        self.evaluate(obj.update_state(y_true, y_pred))
-        self.assertAllClose(0.166666666, obj.result())
+def reset_obj_states(obj1, obj2, obj3):
+    obj1.reset_states()
+    obj2.reset_states()
+    obj3.reset_states()
 
-    def test_with_sparse_labels(self):
-        y_true = np.array([4, 4, 3, 4], dtype=np.int32)
-        y_pred = np.array([4, 4, 1, 2], dtype=np.int32)
 
-        obj = CohenKappa(num_classes=5, sparse_labels=True)
-        self.evaluate(tf.compat.v1.variables_initializer(obj.variables))
+def check_results(objs, values):
+    obj1, obj2, obj3 = objs
+    val1, val2, val3 = values
 
-        self.evaluate(obj.update_state(y_true, y_pred))
-        self.assertAllClose(0.19999999, obj.result())
+    np.testing.assert_allclose(val1, obj1.result(), atol=1e-5)
```

Comment:
Shouldn't we check for a higher tolerance value, like 1e-6 or 1e-7?

Reply:
Possibly, I kept the tolerances used previously. Maybe that can be done in another pull request?

Reply:
Sure. Let's check for higher in the next one